scripts/remove_after_use/divorce_quickfiles.py

```diff
@@ -0,0 +1,106 @@
+from osf.models import QuickFolder, OSFUser, UserLog
+from osf.models.legacy_quickfiles import QuickFilesNode
+from django.core.paginator import Paginator
+from django.db.models import OuterRef, Subquery, Count
+from django.contrib.contenttypes.models import ContentType
+
+import logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def transfer_logs(quickfiles_node):
+    """
+    Recasts logs and saves them, old logs will be deleted with all quickfilesnodes.
+    :param quickfiles_node: a QuickFilesNode
+    :return:
+    """
+    for log in quickfiles_node.logs.all():
+        log.__class__ = UserLog
+        log.save()
+
+
+def create_quickfolders():
+    """
+    Bulk creates a Quickfolder for every user.
+    :return:
+    """
+    users = OSFUser.objects.all()
+    user_content_type_id = ContentType.objects.get_for_model(OSFUser).id
+    quickfolder_content_type_id = ContentType.objects.get_for_model(QuickFolder).id
+
+    paginated_users = Paginator(users, 1000)
+    logger.info('There are {} '.format(users.count()))
+
+    total_created = 0
+    quickfolders_to_create = []
+    for page_num in paginated_users.page_range:
+        for user in paginated_users.page(page_num).object_list:
+            quickfolder = QuickFolder(target_object_id=user.id,
+                                      target_content_type_id=user_content_type_id,
+                                      provider='osfstorage',
+                                      path='/')
+
+            quickfolders_to_create.append(quickfolder)
+
+            total_created += 1
+    QuickFolder.objects.bulk_create(quickfolders_to_create)
+    logger.info('There are {} total quickfolders created'.format(total_created))
+
+
+def repoint_guids():
+    """
+    This takes Guids from Quickfilesnode and repoints them at Quickfolders
+    :return:
+    """
+    guids_repointed = 0
+
+    quickfiles_nodes = QuickFilesNode.objects.all()
+    paginated_quickfiles_nodes = Paginator(quickfiles_nodes, 1000)
+    quickfolder_content_type_id = ContentType.objects.get_for_model(QuickFolder).id
+
+    for page_num in paginated_quickfiles_nodes.page_range:
+        for quickfiles_nodes in paginated_quickfiles_nodes.page(page_num).object_list:
+            guid = quickfiles_nodes.guids.last()
+            guid.referent = guid.referent.creator.quickfolder
+            guid.object_id = guid.referent.target.quickfolder.id
+            guid.content_type_id = quickfolder_content_type_id
+            guid.save()
+            guids_repointed += 1
+
+    logger.info('There are {} total guids repointed to quickfolders'.format(guids_repointed))
+
+
+def migrate_quickfiles_to_quickfolders():
```

Comment:
This should be a data migration, not a temporary cleanup script.  We're doing a fairly significant schema change, so the appropriate place to clean this up is in a data migration.  `remove_after_use` is often to fix bad data, for example.

It needs a forwards and backwards migration, where the backwards migration will undo all of these operations, putting all the user's files back onto the QuickFileNodes instead of QuickFolders.